app/services/remove_status_service.rb

```diff
@@ -38,15 +40,48 @@ def remove_from_followers
     end
   end
 
-  def remove_from_mentioned(stream_entry)
-    salmon_xml       = stream_entry_to_xml(stream_entry)
-    target_accounts  = @mentions.map(&:account).reject(&:local?).uniq(&:domain)
+  def remove_from_remote_mentioned
```

Comment:
remote_mentioned should either be renamed to remote_reblogged_or_mentioned or this should be split into two methods.